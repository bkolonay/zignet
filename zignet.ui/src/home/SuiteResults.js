import React, { Component } from 'react';
import SuiteResultChart from './SuiteResultChart'
import GroupSuitesButton from './GroupSuitesButton'

class SuiteResults extends Component {
  render() {
    return (
  	  <div className="container">
        <div className="text-right">
          <GroupSuitesButton />
        </div>
  	  	<div className="row">
          {this.props.suiteResults.map((suiteResult) =>
            <SuiteResultChart key={suiteResult.SuiteIds[0]}
                              suiteResult={suiteResult} />
          )}
        </div>
      </div>
    );
  }
}

export default SuiteResults;
