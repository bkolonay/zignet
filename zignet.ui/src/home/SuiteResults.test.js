import React from 'react';
import ReactDOM from 'react-dom';

import SuiteResults from './SuiteResults';
import renderer from 'react-test-renderer';

it('renders single suite result', () => {
  const suiteResults = 
    [{
      'SuiteID':1,
      'SuiteName':'suite-name',
      'TotalPassedTests':1,
      'TotalFailedTests':10,
      'TotalInconclusiveTests':0,
      'SuiteEndTime':'2018-02-28T15:05:00'
    }];
  
  ReactDOM.render(
    <SuiteResults suiteResults={suiteResults} />,
    document.createElement('div')
  );
});

it('renders zero suite results', () => {
  const suiteResults = [];
  
  ReactDOM.render(
    <SuiteResults suiteResults={suiteResults} />,
    document.createElement('div')
  );
});

it('renders multiple suite results', () => {
  const suiteResults = 
    [
      {
        'SuiteID':1,
        'SuiteName':'suite-name',
        'TotalPassedTests':1,
        'TotalFailedTests':10,
        'TotalInconclusiveTests':0,
        'SuiteEndTime':'2018-02-28T15:05:00'
      },
      {
        'SuiteID':2,
        'SuiteName':'suite-name 2',
        'TotalPassedTests':9,
        'TotalFailedTests':24,
        'TotalInconclusiveTests':3,
        'SuiteEndTime':'2019-03-28T16:19:00'
      }
    ];
  
  ReactDOM.render(
    <SuiteResults suiteResults={suiteResults} />,
    document.createElement('div')
  );
});

it('matches the previous snapshot', () => {
  const suiteResults = 
    [
      {
        'SuiteID':1,
        'SuiteName':'suite-name',
        'TotalPassedTests':1,
        'TotalFailedTests':10,
        'TotalInconclusiveTests':0,
        'SuiteEndTime':'2018-02-28T15:05:00'
      },
      {
        'SuiteID':2,
        'SuiteName':'suite-name 2',
        'TotalPassedTests':9,
        'TotalFailedTests':24,
        'TotalInconclusiveTests':3,
        'SuiteEndTime':'2019-03-28T16:19:00'
      }
    ];

  const tree = renderer
    .create(<SuiteResults suiteResults={suiteResults} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});