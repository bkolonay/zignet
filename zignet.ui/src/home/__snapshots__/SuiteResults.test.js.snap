// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <h3
        className="text-center"
      >
        suite-name
      </h3>
      <div
        className="ct-chart"
        id="chart1"
      />
      <p
        className="text-center chart-label"
      />
      <p
        className="text-center text-muted"
      >
        <small>
          2 months ago
        </small>
      </p>
    </div>
    <div
      className="col-4"
    >
      <h3
        className="text-center"
      >
        suite-name 2
      </h3>
      <div
        className="ct-chart"
        id="chart2"
      />
      <p
        className="text-center chart-label"
      />
      <p
        className="text-center text-muted"
      >
        <small>
          in a year
        </small>
      </p>
    </div>
  </div>
</div>
`;
